src/Boot/src/AbstractKernel.php

```diff
@@ -278,21 +287,31 @@ public function serve(): mixed
         $eventDispatcher = $this->getEventDispatcher();
         $eventDispatcher?->dispatch(new Serving());
 
+        $serving = $servingScope = null;
         foreach ($this->dispatchers as $dispatcher) {
-            if ($dispatcher->canServe()) {
-                return $this->container->runScope(
-                    [DispatcherInterface::class => $dispatcher],
-                    static function () use ($dispatcher, $eventDispatcher): mixed {
-                        $eventDispatcher?->dispatch(new DispatcherFound($dispatcher));
-                        return $dispatcher->serve();
-                    }
-                );
+            $reflection = new \ReflectionClass($dispatcher);
+
+            $scope = ($reflection->getAttributes(DispatcherScope::class)[0] ?? null)?->newInstance()->scope;
+            $this->container->getBinder($scope)->bind($dispatcher, $dispatcher);
+
+            if ($serving === null && $this->canServe($reflection)) {
```

Comment:
@butschster We can't do this because we bind all dispatchers in the container from the $this->dispatchers property. Therefore, the loop cannot be interrupted.